charmhelpers/contrib/database/mysql.py

```diff
@@ -471,6 +471,18 @@ def configure_db(self, hostname, database, username, admin=False):
         return password
 
 
+# `_singleton_config_helper` stores the instance of the helper class that is
+# being used during a hook invocation.
+_singleton_config_helper = None
+
+
+def get_mysql_config_helper(cls):
+    global _singleton_config_helper
+    if _singleton_config_helper is None:
+        _singleton_config_helper = cls()
+    return _singleton_config_helper
```

Comment:
I wouldn't pass the class in it means the caller could mess with the singleton.  Also, once set it would no longer have a value.  I'd just code it as:

```python
def get_mysql_config_helper():
    global _singleton_config_helper
    if _singleton_config_helper is None:
        _singleton_config_helper = MySQLConfigHelper()
    return _singleton_config_helper
```